hummingbot/market/mock_api_order_book_data_source.py

```diff
@@ -0,0 +1,165 @@
+        # Get the currently active markets
+        trading_pairs: List[str] = await self.get_trading_pairs()
+        retval: Dict[str, OrderBookTrackerEntry] = {}
+
+        number_of_pairs: int = len(trading_pairs)
+        for index, trading_pair in enumerate(trading_pairs):
+            try:
+                snapshot: Dict[str, Any] = await self.get_snapshot(self._client, trading_pair)
+                snapshot_msg: OrderBookMessage = HuobiOrderBook.snapshot_message_from_exchange(
+                    snapshot,
+                    metadata={"symbol": trading_pair}
+                )
+                order_book: OrderBook = self.order_book_create_function()
+                order_book.apply_snapshot(snapshot_msg.bids, snapshot_msg.asks, snapshot_msg.update_id)
+                retval[trading_pair] = OrderBookTrackerEntry(trading_pair, snapshot_msg.timestamp, order_book)
+                self.logger().info(f"Initialized order book for {trading_pair}. "
+                                   f"{index + 1}/{number_of_pairs} completed.")
+                # Huobi rate limit is 100 https requests per 10 seconds
+                await asyncio.sleep(0.4)
+            except Exception:
+                self.logger().error(f"Error getting snapshot for {trading_pair}. ", exc_info=True)
+                await asyncio.sleep(5)
+        return retval
+
+    async def _inner_messages(self,
+                              ws: websockets.WebSocketClientProtocol) -> AsyncIterable[str]:
+        # Terminate the recv() loop as soon as the next message timed out, so the outer loop can reconnect.
+        try:
+            while True:
+                try:
+                    msg: str = await asyncio.wait_for(ws.recv(), timeout=self.MESSAGE_TIMEOUT)
+                    yield msg
+                except asyncio.TimeoutError:
+                    try:
+                        pong_waiter = await ws.ping()
+                        await asyncio.wait_for(pong_waiter, timeout=self.PING_TIMEOUT)
+                    except asyncio.TimeoutError:
+                        raise
+        except asyncio.TimeoutError:
+            self.logger().warning("WebSocket ping timed out. Going to reconnect...")
+            return
+        except ConnectionClosed:
+            return
+        finally:
+            await ws.close()
+
+    async def listen_for_trades(self, ev_loop: asyncio.BaseEventLoop, output: asyncio.Queue):
+        pass
+
+    def inject_mock_diff_message(self, msg: Dict[str, Any]):
+        self._diff_messages.put_nowait(msg)
+
+    def inject_mock_snapshot_message(self, msg: Dict[str, Any]):
+        self._snapshot_messages.put_nowait(msg)
+
+    async def listen_for_order_book_diffs(self, ev_loop: asyncio.BaseEventLoop, output: asyncio.Queue):
+        while True:
+            msg = await self._diff_messages.get()
+            order_book_message: OrderBookMessage = HuobiOrderBook.diff_message_from_exchange(msg)
```

Comment:
It's better to make this more generic and exchange agnostic. Instead of parsing the message with Huobi specific logic, you should pass in a standard order book diff message.